core/src/main/scala/caliban/parsing/Parser.scala

```diff
@@ -266,6 +267,161 @@ object Parser {
         )
     }
 
+  private def schemaExtensionWithDirectivesAndOptionalOperations[_: P]: P[SchemaExtension] =
+    P(directives ~ ("{" ~ rootOperationTypeDefinition.rep ~ "}").?).map {
+      case (directives, Some(ops)) =>
+        val opsMap = ops.toMap
+        SchemaExtension(
+          directives,
+          opsMap.get(OperationType.Query).map(_.name),
+          opsMap.get(OperationType.Mutation).map(_.name),
+          opsMap.get(OperationType.Subscription).map(_.name)
+        )
+      case (directives, None) =>
+        SchemaExtension(
+          directives,
+          None,
+          None,
+          None
+        )
+    }
+
+  private def schemaExtensionWithOperations[_: P]: P[SchemaExtension] =
+    P("{" ~ rootOperationTypeDefinition.rep ~ "}").map { ops =>
+      val opsMap = ops.toMap
+      SchemaExtension(
+        Nil,
+        opsMap.get(OperationType.Query).map(_.name),
+        opsMap.get(OperationType.Mutation).map(_.name),
+        opsMap.get(OperationType.Subscription).map(_.name)
+      )
+    }
+
+  private def schemaExtension[_: P]: P[SchemaExtension] =
+    P("extend schema" ~/ (schemaExtensionWithDirectivesAndOptionalOperations | schemaExtensionWithOperations))
```

Comment:
How about sticking to the spec definitions and have:
- operations + optional directives
- no operations, directives

Reply:
Similarly to the above case, my first version was more like the spec definitions:

```scala
private def schemaExtensionWithOptionalDirectivesAndOperations[_: P]: P[SchemaExtension] =
    P(directives.? ~ "{" ~ rootOperationTypeDefinition.rep ~ "}").map {
      case (directives, ops) =>
        val opsMap = ops.toMap
        SchemaExtension(
          directives.getOrElse(Nil),
          opsMap.get(OperationType.Query).map(_.name),
          opsMap.get(OperationType.Mutation).map(_.name),
          opsMap.get(OperationType.Subscription).map(_.name)
        )
    }

  private def schemaExtensionWithDirectives[_: P]: P[SchemaExtension] =
    P(directives).map(SchemaExtension(_, None, None, None))

  private def schemaExtension[_: P]: P[SchemaExtension] =
    P("extend schema" ~/ (schemaExtensionWithOptionalDirectivesAndOperations | schemaExtensionWithDirectives))
```

Reply:
However, this didn't work for all cases, I got an error for one of the unit tests:

```
extend schema with directives
[info]     Fiber failed.
[info]     A checked error was not handled.
[info]     Parsing Error: Expected document:1:1 / schemaExtension:1:1 / schemaExtensionWithOptionalDirectivesAndOperations:1:15 / "{":1:30, found "" 
```

Reply:
I tried other combinations that were more similar to the specs too, but none of them worked